src/components/multiselect/Multiselect.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { FC, useRef, useState } from "react";
+import IconExpand from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/ExpandMore.svg";
+import { useMultipleSelection, useSelect } from "downshift";
+import clsx from "clsx";
+import { useClickAway, useDeepCompareEffect } from "react-use";
+import CheckBox from "../checkbox/Checkbox";
+import {
+  MultiselectExternalUpdate,
+  MultiselectOption
+} from "../../core/utils/types/multiselect-types";
+import useMultiselectOptions from "./useMultiselectOptions";
+import { useText } from "../../core/utils/text";
+
+export type MultiselectProps = {
+  dataCy?: string;
+  caption?: string;
+  options: MultiselectOption[];
+  defaultValue?: MultiselectOption[];
+  updateExternalState?: MultiselectExternalUpdate;
+};
+
+const Multiselect: FC<MultiselectProps> = ({
+  dataCy = "multiselect",
+  caption,
+  options,
+  defaultValue = [],
+  updateExternalState
+}) => {
+  const t = useText();
+  const ref = useRef(null);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const updateState = (
+    updateKey: string | undefined,
+    value: MultiselectOption[]
+  ) => {
+    if (!updateExternalState || !updateKey) return;
+    updateExternalState.externalUpdateFunction({
+      key: updateKey,
+      value
+    });
+  };
+
+  const { allOptions } = useMultiselectOptions(options, {
+    item: "advancedSearchFilterAllText",
+    value: "all"
+  });
+  const initialSelectedOptions =
+    defaultValue.length > 0 ? defaultValue : allOptions.slice(0, 1);
+
+  const { getDropdownProps, setSelectedItems, selectedItems } =
+    useMultipleSelection({ initialSelectedItems: initialSelectedOptions });
+
+  const addNewSelectedItem = (
+    allCurrentlySelected: MultiselectOption[],
+    newSelected: MultiselectOption,
+    allPossibleOptions: number
+  ) => {
+    // If new selection is not "all" we make sure "all" is deselected
+    if (
+      allCurrentlySelected.find((item) => item.value === "all") &&
+      newSelected.value !== "all"
+    ) {
+      const newValue = [
+        ...selectedItems.filter((item) => item.value !== "all"),
+        newSelected
+      ];
+      updateState(updateExternalState?.key, newValue);
+      setSelectedItems(newValue);
+      return newValue;
+    }
+
+    // If every non-"all" item is selected, then we just select "all"
+    if (
+      newSelected.value !== "all" &&
+      [...allCurrentlySelected, newSelected].length === allPossibleOptions - 1
+    ) {
+      const newValue = [{ item: "advancedSearchFilterAllText", value: "all" }];
+      updateState(updateExternalState?.key, newValue);
+      setSelectedItems(newValue);
+      return newValue;
+    }
```

Comment:
I think it would be nice to separate these actions into functions in a helper file like:
```
selectAllOptions
deselectOtherOptions
...
```
The `updateState` could be a callback that you give the functions